sysinit/coqinit.ml

```diff
@@ -149,10 +149,8 @@ let init_runtime opts =
   List.iter Mltop.add_ml_dir ml_load_path;
   List.iter Loadpath.add_vo_path vo_load_path;
   let ocamlpathsep = if Sys.unix then ":" else ";" in
-  let ocamlpath = String.concat ocamlpathsep ml_load_path in
-  let env_ocamlpath =
-    try Sys.getenv "OCAMLPATH" ^ ocamlpathsep ^ ocamlpath
-    with Not_found -> ocamlpath in
+  let env_ocamlpath = try Sys.getenv "OCAMLPATH" :: ml_load_path with Not_found -> ml_load_path in
+  let env_ocamlpath = String.concat ocamlpathsep (List.rev env_ocamlpath) in
```

Comment:
As mentioned in the issue, using `List.rev` is an unfortunate choice, since it causes `-I` directories to be scanned right-to-left, while the vast majority of compilers (e.g., `gcc`, `ocamlc`) use a left-to-right policy.

- "Directories specified with -I options are scanned in left-to-right order."
- "Directories added with -I are searched [...] in the order in which they were given on the command line"